invenio_records_resources/resources/files/resource.py

```diff
@@ -174,6 +176,32 @@ def read_content(self):
 
         return item.send_file(), 200
 
+    @request_view_args
+    def read_archive(self):
+        """Read a zipped version of all files."""
+        id_ = resource_requestctx.view_args["pid_value"]
+        files = self.service.list_files(
+            g.identity,
+            id_,
+        )
+
+        if files is None:
+            abort(404)
+
+        zs = ZipStream(compress_type=ZIP_STORED)
+        for entry in files.to_dict()["entries"]:
+            file = self.service.get_file_content(g.identity, id_, entry["key"])
+            zs.add(file.get_stream("rb"), entry["key"])
```

Comment:
major: `get_stream` does not close the fp. See [comment](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/records/api.py#L88). Therefore, this would result in a mem leak. Either use `file.open_stream` (seem to recall there was a reason why it wasn't possible) or close after adding?

Reply:
yes, I thought so, too. When I didn't find a solution I analysed what happens and didn't find a clue the files really are left open.

Reply:
Using a context manager (like file.open_stream does) leads to the problem that the files are closed to early and they can't be streamed to the zip.

Reply:
Ok, I was rechecking and you're right: the files are open! So there is a mem leak and I don't know how to get rid of it.

Reply:
The main issue comes from the fact that the files are streamed to the zip while the response is sent right? In that case, if we need them open till the response is fully sent... I guess we need either a task/callback/descriptor method to close them, not fully sure how to implement it tho... Maybe Alex or someone else has better insights :/